### PR TITLE
Feature/event changes

### PR DESCRIPTION
**Pull request by samleeflang**
- Added fields for annotation status, these needs to capture who approved/rejected the annotation
- Fixes to the annotation object 
- Added examples for annotation object
- Changes to the createUpdateTombstoneEvent object so we can capture multiple agents

### CODE REVIEW

**Review by wouteraddink — COMMENTED** [review]
waiting for the latest changes in annotationStatus

File: data-model/fdo-type/annotation/0.3.0/schema/annotation.json
Comment (southeo) [inline data-model/fdo-type/annotation/0.3.0/schema/annotation.json:393]: Why not capture if a decision is pending? If a machine wants to find if a decision has been made, it would have to do a null check. 
something like `ods:DecisionPendingForMerging`
`

**Review by southeo — APPROVED** [review]
:green_salad: 

One comment but I assume it's something that was already discussed

File: data-model/fdo-type/annotation/0.3.0/schema/annotation.json
Comment (samleeflang) [reply data-model/fdo-type/annotation/0.3.0/schema/annotation.json:327]: hmm yes you might be right, so when we create an annotation with motivation adding/editing we will fill this status with DecisionPending. We can then easily filter all annotations which require a decision

File: data-model/fdo-type/annotation/0.3.0/examples/tombstoned-annotation.json
Comment (sharifX) [inline data-model/fdo-type/annotation/0.3.0/examples/tombstoned-annotation.json:43]: TombstonedByAgent should lower case? -> tombstonedByAgent?

File: data-model/fdo-type/annotation/0.3.0/schema/annotation.json
Comment (sharifX) [inline data-model/fdo-type/annotation/0.3.0/schema/annotation.json:274]: yes. I think is better to have the agent as a separate object.

File: data-model/fdo-type/shared-model/0.3.0/schema/tombstone-metadata.json
Comment (southeo) [reply data-model/fdo-type/shared-model/0.3.0/schema/tombstone-metadata.json:1]: Really happy we're talking about tombstoning, but we should think about the fdo profiles alongside these kinds of changes to the data model, or else we'll in for future headaches. :)

File: data-model/fdo-type/source-system/0.3.0/schema/source-system.json
Comment (southeo) [inline data-model/fdo-type/source-system/0.3.0/schema/source-system.json:73]: can we use the `schema` namespace and then point to an `ods` object (agent)? This doesn't seem to me like it's an instance of `schema:creator`

File: data-model/fdo-type/shared-model/0.3.0/schema/citation.json
Comment (southeo) [inline data-model/fdo-type/shared-model/0.3.0/schema/citation.json:34]: if we're not using the dcterms definition anymore, is it still in the dcterms namepsace?

File: data-model/fdo-type/shared-model/0.3.0/schema/agent.json
Comment (southeo) [inline data-model/fdo-type/shared-model/0.3.0/schema/agent.json:7]: Throughout the data model, we call instances of this schema: 

- dcterms:creator
- as:generator
- schema:maintainer
- ods:MergingStateChangedBy
- schema:creator
- ods:hasAgent (arrays of agent)

In my opinion, only the latter is an appropriate usage. I don't think we can use other namespaces to point to an ods object.

We can capture the specifics of an agent, such as `MergingStateChangedBy` or `creator`, using the `schema:roleName` field in this object.

Comment (southeo) [reply data-model/fdo-type/shared-model/0.3.0/schema/agent.json:7]: i have a few comments with this same idea but I can't find them, sorry for repeating myself here

**Review by southeo — CHANGES_REQUESTED** [review]
:corn: 
Few things I think need to be addressed, if not changed, before this can be merged. (sorry for the big red x, i want to undo my previous approval...)

File: data-model/fdo-type/annotation/0.3.0/schema/annotation.json
Comment (TomDijkema) [inline data-model/fdo-type/annotation/0.3.0/schema/annotation.json:271]: Shouldn't this be called 'hasAgent' then?

File: data-model/fdo-type/annotation/0.3.0/schema/annotation.json
Comment (samleeflang) [reply data-model/fdo-type/annotation/0.3.0/schema/annotation.json:327]: Hmm it should be there, just checked and I can see.
```
      "enum": [
        "ods:DecisionPendingForMerging",
        "ods:RejectedForMerging",
        "ods:ApprovedForMerging"
      ],
```

File: data-model/fdo-type/annotation/0.3.0/examples/active-annotation-example.json
Comment (samleeflang) [reply data-model/fdo-type/annotation/0.3.0/examples/active-annotation-example.json]: No Active is the status of the FDO Object, Apporved, which is now `ods:ApprovedForMerging` is a status that the annotation is approved for merging into the original source object.

File: data-model/fdo-type/create-update-tombstone-event/0.3.0/examples/update-event-example.json
Comment (TomDijkema) [inline data-model/fdo-type/create-update-tombstone-event/0.3.0/examples/update-event-example.json:64]: Why does this not point towards the agent object?

File: data-model/fdo-type/annotation/0.3.0/examples/approved-annotation-example.json
Comment (samleeflang) [reply data-model/fdo-type/annotation/0.3.0/examples/approved-annotation-example.json]: We don't use prov in the annotation object. prov-o is not a open set of terms such as schema's, dwc, dcterms but has a fixed structure. We cannot use a term from the namespace on any object.

File: data-model/fdo-type/annotation/0.3.0/examples/tombstoned-annotation.json
Comment (samleeflang) [reply data-model/fdo-type/annotation/0.3.0/examples/tombstoned-annotation.json:43]: Yes that is what we discussed in as namingconvention, if we create new terms and they contain an object we capitalize them to indicate that it contains an object, in this case an agent.

File: data-model/fdo-type/annotation/0.3.0/schema/annotation.json
Comment (samleeflang) [reply data-model/fdo-type/annotation/0.3.0/schema/annotation.json:274]: :+1:

**Review by TomDijkema — COMMENTED** [review]
Nice work! Two questions

File: data-model/fdo-type/annotation/0.3.0/schema/annotation.json
Comment (samleeflang) [reply data-model/fdo-type/annotation/0.3.0/schema/annotation.json:30]: Good point, I would be in favour of reducing this to DRAFT, ACTIVE, TOMBSTONE, TEST and put the reason why it was tombstoned in a different place. Now this is a bit mixed up, we have 5 statuses that indicate it was tombstoned.

File: data-model/fdo-type/shared-model/0.3.0/schema/tombstone-metadata.json
Comment (samleeflang) [reply data-model/fdo-type/shared-model/0.3.0/schema/tombstone-metadata.json:1]: Yes, the term tombstonePids I didn't get and putting just a list of string in there without any context seemed wrong. In the FDO profile the status might indicate a relationship there, but I would be in favour of change the fdo profile at this point. The `ods:tombstoneReason` I will change to `tombstoneText` to keep in line.

File: data-model/fdo-type/annotation/0.3.0/schema/annotation.json
Comment (southeo) [reply data-model/fdo-type/annotation/0.3.0/schema/annotation.json:327]: yep I see it, looks good to me!

File: data-model/fdo-type/source-system/0.3.0/schema/source-system.json
Comment (samleeflang) [reply data-model/fdo-type/source-system/0.3.0/schema/source-system.json:73]: Yes in the instance of `schema` we can actually do it as they leave it open-ended. In addition, the type of the object is not from the `ods` vocab. I you look in the agent you will see that we have five types defined there (where the `prov` types are only used in the provenance object).

File: data-model/fdo-type/shared-model/0.3.0/schema/agent.json
Comment (samleeflang) [reply data-model/fdo-type/shared-model/0.3.0/schema/agent.json:7]: Ok, first thing is that we don't actually create objects of type `ods:Agent` but we create object with a couple types one being `schema:Person`. Schema is pretty open so we can use this in `schema:maintainer` and `schema:creator`. `as:generator` doesn't enforce a specific value type for the object https://www.w3.org/TR/activitystreams-vocabulary/#dfn-generator and neither does dcterms:creator (they recommend an URI, but doesn't say the value type is a string). So I actually don't think we go against any 'rules' when we do it this way.
The second point to include it all in one array and then use `schema:roleName` is a bit tricky as some of the ontologies we use do describe how the object need to look like. So the w3c web annotation model does indicate which terms need to be at what level to improve interoperability. So we can't just remove things such as dcterms:creator and as:generator from these objects.
Anyway that is how I interpreted the termsand when/what to reuse. Might be good if we could find an expert on this to have them check the model and see if we made any mistakes.

File: data-model/fdo-type/create-update-tombstone-event/0.3.0/examples/update-event-example.json
Comment (samleeflang) [reply data-model/fdo-type/create-update-tombstone-event/0.3.0/examples/update-event-example.json:64]: It does, this is a valid object. The agent-object is just a fixed set of properties we use for all places where we refer to agents.

Comment (TomDijkema) [reply data-model/fdo-type/create-update-tombstone-event/0.3.0/examples/update-event-example.json:64]: My bad its an example :0

File: data-model/fdo-type/annotation/0.3.0/schema/annotation.json
Comment (wouteraddink) [reply data-model/fdo-type/annotation/0.3.0/schema/annotation.json:327]: That would work, It looks like as end status for the annotation we really are capturing the mergin decision status, We could then also rename ods:mergingStatus to ods:mergingDecisionStatus with as enum: ods:pending, ods:rejected, ods:approved.

I wonder if we need an extra state for the object ods:status:merged? That may make it also easier for the front-end to decide whether to show an annotation. By default we may not want to show merged annotations with the digital specimen anymore. Or should they be tombstoned after merging? I think they are not active anymore as they cannot be further decided on for merging and should not be commented on further.  I guess annotations that were rejected for merging,  would also be tombstoned.

Comment (samleeflang) [reply data-model/fdo-type/annotation/0.3.0/schema/annotation.json:327]: I agree, let's simplify, and call it `ods:mergingDecisionStatus`. I will push the changes in a moment.

I would say they get tombstoned, the annotation is no longer necessary as it has now become part of the object. After merging we can tombstone the annotation based on the `ods:mergingDecisionStatus` somebody can check if it was approved or rejected.

File: data-model/fdo-type/annotation/0.3.0/schema/annotation.json
Comment (wouteraddink) [reply data-model/fdo-type/annotation/0.3.0/schema/annotation.json:30]: These are valid pid statuses but yes I we could reduce the number of pid statuses by adding a TOMBSTONED status and have the rest as tombstone reasons. RETIRED is a status where you do not want to show a tombstone at all, as if the pid was never there. Think of a lawsuit situation where someone requests a pid to be no longer accessible. But for our use cases I think it is a bit hypothetical and we could just show a tombstone in all cases. Failed is if the pid string was created, but something went wrong with creating the pid metadata. it should not be tombstoned but pid creation should be re-attempted by the pid infrastructure. Depending on how the pid infrastructure was created we may not actually need it, Sou would know better.

Comment (wouteraddink) [reply data-model/fdo-type/annotation/0.3.0/schema/annotation.json:30]: I think we did not do it because it is not really the pid which gets tombstoned but the object that the pid refers to.

File: data-model/fdo-type/shared-model/0.3.0/schema/tombstone-metadata.json
Comment (wouteraddink) [reply data-model/fdo-type/shared-model/0.3.0/schema/tombstone-metadata.json:1]: tombstonePids is an array of pids to show to the user where he can go when the object is tombstoned. the context can then put in the tombstoneText and follows partly from pidStatus but I think that could be improved.

File: data-model/fdo-type/shared-model/0.3.0/schema/agent.json
Comment (wouteraddink) [reply data-model/fdo-type/shared-model/0.3.0/schema/agent.json:7]: it sounds ok to me. Otherwise we end up with having to use ods terms instead and decisions that do not improve interoperability.

**Review by wouteraddink — CHANGES_REQUESTED** [review]
I would add an example or a rejected annotation, as it is now not clear if that would change the ods:status (see my earlier comments). I assume this will remain ods:Active. 

Added fields for annotation status look good and also the changes to the createUpdateTombstoneEvent

File: data-model/fdo-type/annotation/0.3.0/schema/annotation.json
Comment (wouteraddink) [reply data-model/fdo-type/annotation/0.3.0/schema/annotation.json]: ok if they get tombstoned then the example needs to be changed as well.

**Comment by samleeflang** [comment]
Added annotation rejected example and updated the approved annotation example to display status = Tombstone

**Review by wouteraddink — APPROVED** [review]

**Review by TomDijkema — APPROVED** [review]
Very nice!

File: data-model/fdo-type/shared-model/0.3.0/schema/agent.json
Comment (southeo) [reply data-model/fdo-type/shared-model/0.3.0/schema/agent.json:7]: @samleeflang  can you document this discussion somewhere, maybe in release notes? Otherwise sounds great

File: data-model/fdo-type/data-mapping/0.3.0/examples/tombstoned-data-mapping-example.json
Comment (sharifX) [inline data-model/fdo-type/data-mapping/0.3.0/examples/tombstoned-data-mapping-example.json:6]: should this be "Tombstoned"?

File: data-model/fdo-type/data-mapping/0.3.0/schema/data-mapping.json
Comment (sharifX) [inline data-model/fdo-type/data-mapping/0.3.0/schema/data-mapping.json:41]: Tombstoned?

File: data-model/fdo-type/digital-media/0.3.0/examples/tombstoned-digital-media-example.json
Comment (sharifX) [inline data-model/fdo-type/digital-media/0.3.0/examples/tombstoned-digital-media-example.json:6]: Here this is ok.

**Review by southeo — APPROVED** [review]
need to think about fdo profiles but that's beyond the scope.

**Comment by sharifX** [comment]
ods:Tombstone is ok to be consistent with the rest of the ods status (Active, Draft)

File: data-model/fdo-type/digital-media/0.3.0/examples/tombstoned-digital-media-example.json
Comment (samleeflang) [reply data-model/fdo-type/digital-media/0.3.0/examples/tombstoned-digital-media-example.json]: Fixed this example

**Pull request merged**
